src/node.rs

```diff
@@ -566,7 +593,27 @@ impl PbftNode {
             return Ok(());
         }
 
-        info!("{}: Got BlockNew: {:?}", state, block.block_id);
+        let head = self
+            .service
+            .get_chain_head()
+            .map_err(|e| PbftError::InternalError(e.description().to_string()))?;
+
+        info!(
+            "{}: Got BlockNew: {} / {}",
+            state,
+            block.block_num,
+            hex::encode(&block.block_id[..6]),
+        );
+
+        debug!("Head is currently at {}", head.block_num);
+
+        if block.block_num <= head.block_num {
```

Comment:
Again, this seems like a fairly self-contained change that isn't properly captured by the commit message. Probably better off with its own commit.